libbeat/outputs/kafka/partition.go

```diff
@@ -153,12 +153,13 @@ func cfgRoundRobinPartitioner(config *common.Config) (func() partitioner, error)
 	return func() partitioner {
 		N := cfg.GroupEvents
 		count := N
-		partition := rand.Int31()
+		var partition int32
+		partition = 0
```

Comment:
the `rand` has been on purpose, such that the first partition to publish too is chosen by random in order to have a chance to somewhat evenly distribute events if N is large and beats get's restarted.